{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "## **Spoiler: no partitioning because of problems with the data**\n",
        "\n",
        "âŒ Steps to achive our goal \"Article reproduction\":\n",
        "\n",
        "1. Download all fasta for genes COI and 18S\n",
        "2. Sequnces alignment separately for 2 genes (MAFFT)\n",
        "3. Check the alignments (Ugene)\n",
        "4. Using Trimal for trimming the alignments if necessary\n",
        "5. Use IQTree to get the right model for each gene\n",
        "6. Using FASconCAT-G to concatenate 2 genes\n",
        "7. Use explanation from [IQTree website](http://www.iqtree.org/doc/Complex-Models) to get the tree with partitioning\n",
        "8. Visualize the tree\n",
        "\n",
        "Additional part:\n",
        "\n",
        "*   Use species only with the rearrangements to get the close up on it (repeat the previous steps but only with this species)\n",
        "*   To visualize the tree add the metadata\n",
        "\n",
        "\n",
        "> I guess it would be better to use different collab motebooks for different parts of the work (\"Downloading data\", \"General tree reconstraction\" and \"Rearrangements tree\")\n",
        "\n",
        "---\n",
        "\n",
        "> We wanted to use partitioning for this task but the problem is data:\n",
        "\n",
        "*   18S: whole genome downloading (too big files) 33/215\n",
        "*   28S: whole genome downloading (too big files) 88/215\n",
        "*   COI: only one which is presented for all species\n",
        "*   ITS1: 16/215\n",
        "*   Cytb: 62/215 mitochondrion, complete genome\n",
        "\n",
        "---\n",
        "\n",
        "**No partitioning then** ðŸ˜ž"
      ],
      "metadata": {
        "id": "sy560Z6epTSI"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## **General tree reconstruction**\n",
        "\n",
        "**Steps to achive our goal \"Article reproduction\":**\n",
        "\n",
        "1. Download all fasta for gene COI (\"Downloading_data\" notebook)\n",
        "2. Sequnces alignment (MAFFT)\n",
        "3. Check the alignments (Ugene)\n",
        "4. Using Trimal for trimming the alignments if necessary (and it's necessary)\n",
        "5. Check outgroup IDs\n",
        "6. Use IQTree to get the tree (outgroup!)\n",
        "7. Rename the tree (\"Downloading_data\" notebook)\n",
        "8. Visualize the tree\n",
        "\n",
        "Additional part:\n",
        "\n",
        "*   Get the metadata\n",
        "*   To visualize the tree add the metadata\n",
        "*   Explain rearrangements"
      ],
      "metadata": {
        "id": "SMHQnO04k3qG"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## **Step 1. Alignment + trimming**\n",
        "\n",
        "We are going to use mafft for sequence alignment and Trimal to trimm it then.\n",
        "\n",
        "> Since we have complete mitochondrial genomes instead of one COI gene for some species this part with trimming is necessary."
      ],
      "metadata": {
        "id": "pIgsfYoAmdkt"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "%%capture\n",
        "!wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n",
        "!bash Miniconda3-latest-Linux-x86_64.sh -bfp /usr/local"
      ],
      "metadata": {
        "id": "U15Hl4jCnek8"
      },
      "execution_count": 1,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**1.1 Sequnces alignment**"
      ],
      "metadata": {
        "id": "wdwZxairnjFV"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "GIhUxpdApBsd"
      },
      "outputs": [],
      "source": [
        "%%capture\n",
        "!apt-get install mafft"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mafft --auto COI_sequences.fasta > COI_aligned.fasta"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "yByC9u85nUoI",
        "outputId": "40195c61-a26a-4246-f966-088231f1c2e2"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "nthread = 0\n",
            "nthreadpair = 0\n",
            "nthreadtb = 0\n",
            "ppenalty_ex = 0\n",
            "stacksize: 8192 kb\n",
            "generating a scoring matrix for nucleotide (dist=200) ... done\n",
            "Gap Penalty = -1.53, +0.00, +0.00\n",
            "\n",
            "\n",
            "\n",
            "Making a distance matrix ..\n",
            "\n",
            "There are 10 ambiguous characters.\n",
            "\r    1 / 215\r  101 / 215\r  201 / 215\n",
            "done.\n",
            "\n",
            "Constructing a UPGMA tree (efffree=0) ... \n",
            "\r    0 / 215\r   10 / 215\r   20 / 215\r   30 / 215\r   40 / 215\r   50 / 215\r   60 / 215\r   70 / 215\r   80 / 215\r   90 / 215\r  100 / 215\r  110 / 215\r  120 / 215\r  130 / 215\r  140 / 215\r  150 / 215\r  160 / 215\r  170 / 215\r  180 / 215\r  190 / 215\r  200 / 215\r  210 / 215\n",
            "done.\n",
            "\n",
            "Progressive alignment 1/2... \n",
            "STEP    86 / 214  f\b\b\n",
            "Reallocating..done. *alloclen = 32988\n",
            "STEP   214 / 214  f\b\b\n",
            "done.\n",
            "\n",
            "Making a distance matrix from msa.. \n",
            "  200 / 215\n",
            "done.\n",
            "\n",
            "Constructing a UPGMA tree (efffree=1) ... \n",
            "  210 / 215\n",
            "done.\n",
            "\n",
            "Progressive alignment 2/2... \n",
            "STEP   189 / 214  f\b\b\n",
            "Reallocating..done. *alloclen = 32863\n",
            "STEP   214 / 214  f\b\b\n",
            "done.\n",
            "\n",
            "disttbfast (nuc) Version 7.490\n",
            "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
            "0 thread(s)\n",
            "\n",
            "\n",
            "Strategy:\n",
            " FFT-NS-2 (Fast but rough)\n",
            " Progressive method (guide trees were built 2 times.)\n",
            "\n",
            "If unsure which option to use, try 'mafft --auto input > output'.\n",
            "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
            "\n",
            "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
            "It tends to insert more gaps into gap-rich regions than previous versions.\n",
            "To disable this change, add the --leavegappyregion option.\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**1.2 Alignment trimming**"
      ],
      "metadata": {
        "id": "qscNabV2nsVg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!conda create -n trimal"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "TBQvGzl1n2_b",
        "outputId": "ac8fb00a-8736-4349-cf80-4601e1cfb742"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local/envs/trimal\n",
            "\n",
            "\n",
            "\n",
            "Proceed ([y]/n)? y\n",
            "\n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\bdone\n",
            "Executing transaction: | \b\bdone\n",
            "#\n",
            "# To activate this environment, use\n",
            "#\n",
            "#     $ conda activate trimal\n",
            "#\n",
            "# To deactivate an active environment, use\n",
            "#\n",
            "#     $ conda deactivate\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!source activate trimal && conda install trimal -c conda-forge -c bioconda"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "gPN-9mhCn5vy",
        "outputId": "8c074477-a424-446e-d919-6711c37fe071"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - conda-forge\n",
            " - bioconda\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local/envs/trimal\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimal\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge\n",
            "    _openmp_mutex-4.5          |            2_gnu          23 KB  conda-forge\n",
            "    libgcc-14.2.0              |       h77fa898_1         829 KB  conda-forge\n",
            "    libgcc-ng-14.2.0           |       h69a702a_1          53 KB  conda-forge\n",
            "    libgomp-14.2.0             |       h77fa898_1         450 KB  conda-forge\n",
            "    libstdcxx-14.2.0           |       hc0a3c3a_1         3.7 MB  conda-forge\n",
            "    libstdcxx-ng-14.2.0        |       h4852527_1          53 KB  conda-forge\n",
            "    trimal-1.5.0               |       h4ac6f70_1         211 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         5.3 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge \n",
            "  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-2_gnu \n",
            "  libgcc             conda-forge/linux-64::libgcc-14.2.0-h77fa898_1 \n",
            "  libgcc-ng          conda-forge/linux-64::libgcc-ng-14.2.0-h69a702a_1 \n",
            "  libgomp            conda-forge/linux-64::libgomp-14.2.0-h77fa898_1 \n",
            "  libstdcxx          conda-forge/linux-64::libstdcxx-14.2.0-hc0a3c3a_1 \n",
            "  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-14.2.0-h4852527_1 \n",
            "  trimal             bioconda/linux-64::trimal-1.5.0-h4ac6f70_1 \n",
            "\n",
            "\n",
            "Proceed ([y]/n)? y\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "libstdcxx-14.2.0     | 3.7 MB    | :   0% 0/1 [00:00<?, ?it/s]\n",
            "libgcc-14.2.0        | 829 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "libgomp-14.2.0       | 450 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "trimal-1.5.0         | 211 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "libgcc-ng-14.2.0     | 53 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libstdcxx-ng-14.2.0  | 53 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "_openmp_mutex-4.5    | 23 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "_libgcc_mutex-0.1    | 3 KB      | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "libstdcxx-14.2.0     | 3.7 MB    | :   0% 0.004207828296772089/1 [00:00<00:26, 26.28s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "_libgcc_mutex-0.1    | 3 KB      | : 100% 1.0/1 [00:00<00:00,  7.19it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libstdcxx-ng-14.2.0  | 53 KB     | :  30% 0.3028185934756492/1 [00:00<00:00,  2.02it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "_openmp_mutex-4.5    | 23 KB     | :  69% 0.6936200838237162/1 [00:00<00:00,  4.42it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "libgcc-ng-14.2.0     | 53 KB     | : 100% 1.0/1 [00:00<00:00,  3.87it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "libgcc-ng-14.2.0     | 53 KB     | : 100% 1.0/1 [00:00<00:00,  3.87it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "libgomp-14.2.0       | 450 KB    | : 100% 1.0/1 [00:00<00:00,  3.46it/s]\u001b[A\u001b[A\n",
            "\n",
            "libgomp-14.2.0       | 450 KB    | : 100% 1.0/1 [00:00<00:00,  3.46it/s]\u001b[A\u001b[A\n",
            "libgcc-14.2.0        | 829 KB    | : 100% 1.0/1 [00:00<00:00,  2.40it/s]               \u001b[A\n",
            "libgcc-14.2.0        | 829 KB    | : 100% 1.0/1 [00:00<00:00,  2.40it/s]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "_libgcc_mutex-0.1    | 3 KB      | : 100% 1.0/1 [00:00<00:00,  7.19it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "trimal-1.5.0         | 211 KB    | : 100% 1.0/1 [00:00<00:00,  2.12it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "trimal-1.5.0         | 211 KB    | : 100% 1.0/1 [00:00<00:00,  2.12it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libstdcxx-ng-14.2.0  | 53 KB     | : 100% 1.0/1 [00:00<00:00,  2.04it/s]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libstdcxx-ng-14.2.0  | 53 KB     | : 100% 1.0/1 [00:00<00:00,  2.04it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\bdone\n",
            "Executing transaction: | \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# COI trimming with -gt 0.1 param\n",
        "!source activate trimal && trimal -in COI_aligned.fasta -out COI_trimmed.fasta -gt 0.1"
      ],
      "metadata": {
        "id": "BwAUiXg6oAlE"
      },
      "execution_count": 7,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!mafft --add /content/modified_mt.fa --reorder /content/COI_trimmed.fasta > /content/add_trimmed.fasta"
      ],
      "metadata": {
        "id": "-ChTMbXIGdpv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "> We lost this one after trimming \"*DQ078424.1 Melinaea marsaeus*\". and that's interesting cause its not inverted or smth seems like just lost the COI part in it."
      ],
      "metadata": {
        "id": "X5swOQ-hsXaM"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## **Step 2. Tree reconstruction**\n",
        "\n",
        "We are going to use IQTree for this and the outgroup (Trichoptera):\n",
        "\n",
        "* HM902392.1\n",
        "* OP817850.1\n",
        "* MZ629096.1\n",
        "* MZ628664.1\n",
        "* MZ627614.1"
      ],
      "metadata": {
        "id": "jCuhkI0Tol7I"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!conda create -n iqtree"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "Lgd0e4_ls8jU",
        "outputId": "88327649-fe82-4269-e43c-2ba781c8658f"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local/envs/iqtree\n",
            "\n",
            "\n",
            "\n",
            "Proceed ([y]/n)? y\n",
            "\n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\bdone\n",
            "Executing transaction: | \b\bdone\n",
            "#\n",
            "# To activate this environment, use\n",
            "#\n",
            "#     $ conda activate iqtree\n",
            "#\n",
            "# To deactivate an active environment, use\n",
            "#\n",
            "#     $ conda deactivate\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!source activate iqtree && conda install iqtree -c conda-forge -c bioconda"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "d2K9RhBzs_Pt",
        "outputId": "d5ab5852-972a-4386-8f33-e7c5cbe8242a"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - conda-forge\n",
            " - bioconda\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local/envs/iqtree\n",
            "\n",
            "  added / updated specs:\n",
            "    - iqtree\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    iqtree-2.3.6               |       hdbdd923_0         4.0 MB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         4.0 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge \n",
            "  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-2_gnu \n",
            "  iqtree             bioconda/linux-64::iqtree-2.3.6-hdbdd923_0 \n",
            "  libgcc             conda-forge/linux-64::libgcc-14.2.0-h77fa898_1 \n",
            "  libgcc-ng          conda-forge/linux-64::libgcc-ng-14.2.0-h69a702a_1 \n",
            "  libgomp            conda-forge/linux-64::libgomp-14.2.0-h77fa898_1 \n",
            "  libstdcxx          conda-forge/linux-64::libstdcxx-14.2.0-hc0a3c3a_1 \n",
            "  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-14.2.0-h4852527_1 \n",
            "\n",
            "\n",
            "Proceed ([y]/n)? y\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "                                                                        \n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!source activate iqtree && iqtree -s COI_trimmed.fasta -m TEST -bb 1000 -nt AUTO -o HM902392.1,OP817850.1,MZ629096.1,MZ628664.1,MZ627614.1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "lQaWVPVfolBT",
        "outputId": "bc1cfc34-cef3-49f7-96d6-5c64b5938e6a"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "IQ-TREE multicore version 2.3.6 for Linux x86 64-bit built Aug  4 2024\n",
            "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor, Heiko Schmidt,\n",
            "Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong\n",
            "\n",
            "Host:    cbb8eb040054 (AVX2, FMA3, 12 GB RAM)\n",
            "Command: iqtree -s COI_trimmed.fasta -m TEST -bb 1000 -nt AUTO -o HM902392.1,OP817850.1,MZ629096.1,MZ628664.1,MZ627614.1\n",
            "Seed:    724479 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
            "Time:    Sun Oct 27 13:49:54 2024\n",
            "Kernel:  AVX+FMA - auto-detect threads (2 CPU cores detected)\n",
            "\n",
            "Reading alignment file COI_trimmed.fasta ... Fasta format detected\n",
            "Reading fasta file: done in 0.00446346 secs using 86.46% CPU\n",
            "Alignment most likely contains DNA/RNA sequences\n",
            "Alignment has 215 sequences with 658 columns, 524 distinct patterns\n",
            "344 parsimony-informative, 162 singleton sites, 152 constant sites\n",
            "             Gap/Ambiguity  Composition  p-value\n",
            "Analyzing sequences: done in 0.000290618 secs using 481.4% CPU\n",
            "   1  HM902392.1     0.00%    passed     52.47%\n",
            "   2  OP817850.1     0.00%    passed     44.84%\n",
            "   3  MZ629096.1     0.91%    passed     17.05%\n",
            "   4  MZ628664.1     0.46%    passed     32.49%\n",
            "   5  MZ627614.1     0.46%    passed     14.32%\n",
            "   6  OQ563739.1     0.00%    passed     99.01%\n",
            "   7  OQ562774.1     0.00%    passed     74.72%\n",
            "   8  OQ181966.1     6.08%    passed     89.63%\n",
            "   9  OQ182072.1     0.00%    passed     79.17%\n",
            "  10  OQ183174.1     0.00%    passed     89.84%\n",
            "  11  OQ564337.1     0.00%    passed     96.34%\n",
            "  12  OQ182916.1     0.00%    passed     78.37%\n",
            "  13  OQ564147.1     0.00%    passed     98.88%\n",
            "  14  OR368811.1     0.00%    passed     88.60%\n",
            "  15  OR614510.1     0.00%    passed     96.83%\n",
            "  16  OQ564221.1     0.00%    passed     99.84%\n",
            "  17  OQ564099.1     0.00%    passed     84.34%\n",
            "  18  OQ182298.1     0.00%    passed     90.47%\n",
            "  19  OQ182974.1     0.00%    passed     73.60%\n",
            "  20  OQ563752.1     0.00%    passed     78.07%\n",
            "  21  OQ181675.1     0.00%    passed     95.88%\n",
            "  22  OQ182040.1     0.00%    passed     99.04%\n",
            "  23  OQ563910.1     0.00%    passed     98.88%\n",
            "  24  OR890456.1     0.15%    passed     81.25%\n",
            "  25  OQ563644.1     0.00%    passed     87.93%\n",
            "  26  OR369297.1     0.00%    passed     48.50%\n",
            "  27  OR890482.1    15.05%    passed     28.76%\n",
            "  28  OR367990.1     0.00%    passed     58.66%\n",
            "  29  PP728018.1     5.17%    passed     81.80%\n",
            "  30  JN266515.1     0.00%    passed     85.41%\n",
            "  31  MN387650.1     0.00%    passed     73.73%\n",
            "  32  MF049944.1     6.53%    passed     89.71%\n",
            "  33  OR890504.1     0.15%    passed     47.39%\n",
            "  34  OR890505.1     5.93%    passed     66.71%\n",
            "  35  OQ564019.1     0.00%    passed     52.68%\n",
            "  36  OQ563365.1     0.00%    passed     60.83%\n",
            "  37  GU655892.1     0.00%    passed     63.74%\n",
            "  38  JN279253.1     0.00%    passed     30.85%\n",
            "  39  OR369609.1     0.00%    passed     88.41%\n",
            "  40  HM405738.1     3.04%    passed     78.05%\n",
            "  41  HM405744.1     2.28%    passed     80.93%\n",
            "  42  OR890530.1     7.45%    passed     65.85%\n",
            "  43  PQ329533.1     0.00%    passed     38.11%\n",
            "  44  OQ181924.1     1.67%    passed     61.52%\n",
            "  45  OR890548.1    20.06%    passed     63.99%\n",
            "  46  OQ563713.1     0.00%    passed     66.05%\n",
            "  47  OR368842.1     0.00%    passed     89.53%\n",
            "  48  OQ564309.1     0.00%    passed     80.28%\n",
            "  49  OQ562769.1     0.00%    passed     63.42%\n",
            "  50  NC_023212.1    0.00%    passed     77.71%\n",
            "  51  OR890569.1     7.90%    passed     87.08%\n",
            "  52  OQ181619.1     0.00%    passed     91.05%\n",
            "  53  OR890571.1     7.60%    passed     77.03%\n",
            "  54  OR369168.1     0.00%    passed     75.87%\n",
            "  55  OQ564235.1     0.00%    passed     93.18%\n",
            "  56  HM914048.1     0.00%    passed     95.92%\n",
            "  57  OR890620.1     6.08%    passed     87.46%\n",
            "  58  OQ563650.1     0.00%    passed     98.70%\n",
            "  59  OQ562624.1     0.00%    passed     84.20%\n",
            "  60  OR369643.1     0.00%    passed     41.80%\n",
            "  61  OR890700.1     0.15%    passed     77.95%\n",
            "  62  OQ563890.1     0.00%    passed     55.11%\n",
            "  63  KP007712.1     0.00%    passed     84.48%\n",
            "  64  OQ563604.1     0.00%    passed     61.67%\n",
            "  65  KY000412.1     0.00%    passed     97.65%\n",
            "  66  NC_027156.1    0.00%    passed     88.04%\n",
            "  67  OQ562126.1     0.00%    passed     76.34%\n",
            "  68  HM871300.1     0.00%    passed     89.55%\n",
            "  69  HQ977510.1     0.00%    passed     84.24%\n",
            "  70  HM876642.1     0.00%    passed     97.15%\n",
            "  71  OR369635.1     0.00%    passed     99.01%\n",
            "  72  MH286153.1     7.14%    passed     64.24%\n",
            "  73  PQ298449.1     0.00%    passed     79.73%\n",
            "  74  OQ562385.1     0.00%    passed     76.11%\n",
            "  75  OQ182417.1     0.00%    passed     99.01%\n",
            "  76  OQ564211.1     0.00%    passed     55.24%\n",
            "  77  OR369411.1     0.00%    passed     74.09%\n",
            "  78  OR368897.1     0.00%    passed     25.68%\n",
            "  79  OQ563635.1     0.00%    passed     84.85%\n",
            "  80  OQ563450.1     0.00%    passed     72.23%\n",
            "  81  OR369281.1     0.00%    passed     93.56%\n",
            "  82  OR890702.1     4.10%    passed     48.28%\n",
            "  83  OR890742.1     6.99%    passed     26.43%\n",
            "  84  OR890752.1     7.45%    passed     93.43%\n",
            "  85  GU654971.1     0.00%    passed     96.43%\n",
            "  86  MN621055.1     0.00%    passed     68.23%\n",
            "  87  OR367879.1     0.00%    passed     72.62%\n",
            "  88  OR369033.1     0.00%    passed     45.66%\n",
            "  89  OR369420.1     0.00%    passed     55.28%\n",
            "  90  OQ564024.1     0.00%    passed     98.61%\n",
            "  91  OQ563661.1     0.00%    passed     58.49%\n",
            "  92  OQ182321.1     0.00%    passed     53.90%\n",
            "  93  OR890766.1     7.60%    passed     88.84%\n",
            "  94  OQ182387.1     0.00%    passed     90.65%\n",
            "  95  OR890773.1     3.80%    passed     27.99%\n",
            "  96  NC_014806.1    0.00%    passed     97.95%\n",
            "  97  OQ562738.1     0.00%    passed     90.69%\n",
            "  98  OR369690.1     0.00%    passed     85.72%\n",
            "  99  OQ563625.1     0.00%    passed     64.03%\n",
            " 100  OQ563057.1     0.00%    passed     88.71%\n",
            " 101  OR614532.1    15.50%    passed     94.78%\n",
            " 102  OY776805.1    42.86%    passed     69.28%\n",
            " 103  MK779614.1     0.00%    passed     79.92%\n",
            " 104  OQ562034.1     0.00%    passed     97.51%\n",
            " 105  OR890786.1    19.30%    passed     64.52%\n",
            " 106  HM377870.1     0.00%    passed     68.77%\n",
            " 107  OR890795.1     0.61%    passed     96.71%\n",
            " 108  JN286946.1    11.70%    passed     60.08%\n",
            " 109  OR369505.1     0.00%    passed     83.38%\n",
            " 110  OR369611.1     0.00%    passed     65.14%\n",
            " 111  OQ564330.1     0.00%    passed     78.96%\n",
            " 112  JN678068.1     0.00%    passed     83.77%\n",
            " 113  OQ563939.1     0.00%    passed     92.21%\n",
            " 114  OQ562780.1     0.00%    passed     34.62%\n",
            " 115  OR368924.1     4.71%    passed     97.74%\n",
            " 116  OR890798.1     7.60%    passed     96.29%\n",
            " 117  OR890801.1     7.45%    passed     63.24%\n",
            " 118  OQ563937.1     0.00%    passed     96.00%\n",
            " 119  OR890817.1     0.30%    passed     78.80%\n",
            " 120  MF054380.1     0.61%    passed     96.65%\n",
            " 121  KX230819.1    41.95%    passed     41.99%\n",
            " 122  OR890821.1     8.36%    passed     94.10%\n",
            " 123  OR890826.1     0.15%    passed     96.86%\n",
            " 124  OR369602.1     0.00%    passed     98.92%\n",
            " 125  OQ564149.1     0.00%    passed     98.11%\n",
            " 126  OR890860.1     2.89%    passed     68.95%\n",
            " 127  PQ298649.1     0.00%    passed     98.21%\n",
            " 128  OR891252.1     2.89%    passed     71.03%\n",
            " 129  OR891256.1     7.45%    passed     72.82%\n",
            " 130  OR368648.1     0.00%    passed     36.71%\n",
            " 131  OQ563675.1     0.00%    passed     88.08%\n",
            " 132  HQ568342.1     3.04%    passed     54.88%\n",
            " 133  OR890957.1    36.63%    passed     73.95%\n",
            " 134  OQ182619.1     0.00%    passed     48.34%\n",
            " 135  HQ953226.1     0.00%    passed     87.02%\n",
            " 136  OQ564348.1     0.00%    passed     75.66%\n",
            " 137  OR890980.1    36.63%    passed     86.51%\n",
            " 138  DQ078424.1    12.92%    failed      0.00%\n",
            " 139  KU892715.1     5.32%    passed     69.75%\n",
            " 140  OR891065.1     0.30%    passed     92.77%\n",
            " 141  OR890992.1     0.30%    passed     98.63%\n",
            " 142  OQ563515.1     0.00%    passed     91.94%\n",
            " 143  OQ182001.1     7.14%    failed      0.14%\n",
            " 144  OQ563808.1     0.00%    passed     87.41%\n",
            " 145  OQ564173.1     0.00%    passed     97.82%\n",
            " 146  OQ564350.1     0.00%    passed     92.79%\n",
            " 147  OQ564303.1     0.00%    passed     94.64%\n",
            " 148  OQ562882.1     0.00%    passed     95.96%\n",
            " 149  MT324064.1     0.00%    passed     51.27%\n",
            " 150  OQ182983.1     0.00%    passed     80.50%\n",
            " 151  OQ563516.1     0.00%    passed     67.46%\n",
            " 152  OQ563874.1     0.00%    passed     85.41%\n",
            " 153  OQ564204.1     0.00%    passed     79.33%\n",
            " 154  OQ564195.1     0.00%    passed     48.91%\n",
            " 155  OQ564004.1     0.00%    passed     25.89%\n",
            " 156  OR891101.1     0.15%    passed     53.40%\n",
            " 157  OR890450.1     1.98%    passed     98.57%\n",
            " 158  OR891109.1     8.21%    passed     67.57%\n",
            " 159  OQ564282.1     0.00%    passed     68.89%\n",
            " 160  MH850976.1     0.00%    passed     97.96%\n",
            " 161  OR614726.1     0.00%    passed     88.24%\n",
            " 162  OR369569.1     0.00%    passed     76.12%\n",
            " 163  OQ562331.1     0.00%    passed     63.62%\n",
            " 164  OQ563254.1     0.00%    passed     99.97%\n",
            " 165  OQ182700.1     0.00%    passed     94.16%\n",
            " 166  OR891111.1     7.45%    passed     86.86%\n",
            " 167  OQ563225.1     0.00%    passed     70.48%\n",
            " 168  OR891117.1    20.06%    passed     79.25%\n",
            " 169  OR369180.1     3.04%    passed     67.73%\n",
            " 170  OQ563925.1     0.00%    passed     98.95%\n",
            " 171  OQ182745.1     0.00%    passed     51.16%\n",
            " 172  OQ563991.1     0.00%    passed     80.16%\n",
            " 173  OR368942.1     0.00%    passed     30.32%\n",
            " 174  OQ564139.1     0.00%    passed     99.53%\n",
            " 175  OQ563986.1     0.00%    passed     98.79%\n",
            " 176  OR891167.1     7.60%    passed     57.08%\n",
            " 177  OR891179.1     7.29%    passed     92.74%\n",
            " 178  OR891202.1     7.90%    passed     87.59%\n",
            " 179  OR891213.1     0.46%    passed     65.49%\n",
            " 180  OQ564031.1     0.00%    passed     97.92%\n",
            " 181  OR891280.1     9.57%    passed     90.98%\n",
            " 182  OQ564246.1     0.00%    passed     96.56%\n",
            " 183  OQ563498.1     0.00%    passed     26.22%\n",
            " 184  OR891316.1     7.45%    passed     84.61%\n",
            " 185  OQ562561.1     0.00%    passed     74.50%\n",
            " 186  OR369507.1     0.00%    passed     68.13%\n",
            " 187  OL534370.1     0.00%    failed      0.14%\n",
            " 188  JN279255.1     0.00%    passed     60.80%\n",
            " 189  OQ564168.1     0.00%    passed     91.71%\n",
            " 190  OQ563098.1     0.00%    passed     98.79%\n",
            " 191  OQ564362.1     0.00%    passed     94.23%\n",
            " 192  ON704401.1    26.14%    failed      0.03%\n",
            " 193  NC_022676.1    0.00%    passed     92.21%\n",
            " 194  HM871659.1     0.00%    passed     94.16%\n",
            " 195  GU706630.1     0.00%    passed     18.72%\n",
            " 196  JN279252.1     0.00%    passed     35.28%\n",
            " 197  KP253407.1     0.00%    passed     17.56%\n",
            " 198  OR369499.1     0.00%    passed     55.31%\n",
            " 199  OQ563584.1     0.00%    passed     85.23%\n",
            " 200  OQ564245.1     0.00%    passed     67.23%\n",
            " 201  OR891397.1     6.84%    passed     70.73%\n",
            " 202  JF848487.1     0.00%    passed     50.90%\n",
            " 203  OQ564107.1     0.00%    passed     76.00%\n",
            " 204  OR891400.1     2.89%    passed     86.57%\n",
            " 205  OR891401.1     0.30%    passed     99.80%\n",
            " 206  OQ563116.1     0.00%    passed     34.71%\n",
            " 207  OQ564124.1     0.00%    passed     95.45%\n",
            " 208  HM872583.1     0.00%    passed     93.54%\n",
            " 209  OQ563838.1     0.00%    passed     97.58%\n",
            " 210  HM405698.1     0.15%    passed     84.19%\n",
            " 211  HQ955534.1     0.00%    passed     36.75%\n",
            " 212  OQ182718.1     0.00%    passed     93.43%\n",
            " 213  OQ183125.1     0.00%    passed     23.57%\n",
            " 214  OQ563627.1     0.00%    passed     57.76%\n",
            " 215  MK931019.1     0.00%    passed     35.75%\n",
            "****  TOTAL          2.46%  4 sequences failed composition chi2 test (p-value<5%; df=3)\n",
            "\n",
            "\n",
            "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.050 seconds\n",
            "Measuring multi-threading efficiency up to 2 CPU cores\n",
            "Increase to 10 rounds for branch lengths\n",
            "7 trees examined\n",
            "Threads: 1 / Time: 2.037 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -43886\n",
            "Threads: 2 / Time: 5.661 sec / Speedup: 0.360 / Efficiency: 18% / LogL: -43886\n",
            "BEST NUMBER OF THREADS: 1\n",
            "\n",
            "Perform fast likelihood tree search using GTR+I+G model...\n",
            "Estimate model parameters (epsilon = 5.000)\n",
            "Perform nearest neighbor interchange...\n",
            "Estimate model parameters (epsilon = 1.000)\n",
            "1. Initial log-likelihood: -35315.063\n",
            "2. Current log-likelihood: -35313.306\n",
            "Optimal log-likelihood: -35313.042\n",
            "Rate parameters:  A-C: 3.19120  A-G: 10.49986  A-T: 14.73314  C-G: 2.38324  C-T: 58.16208  G-T: 1.00000\n",
            "Base frequencies:  A: 0.304  C: 0.158  G: 0.146  T: 0.392\n",
            "Proportion of invariable sites: 0.183\n",
            "Gamma shape alpha: 0.475\n",
            "Parameters optimization took 2 rounds (0.504 sec)\n",
            "Time for fast ML tree search: 4.548 seconds\n",
            "\n",
            "NOTE: ModelFinder requires 16 MB RAM!\n",
            "ModelFinder will test up to 88 DNA models (sample size: 658 epsilon: 0.100) ...\n",
            " No. Model         -LnL         df  AIC          AICc         BIC\n",
            "  1  GTR+F         43564.694    435 87999.388    89708.037    89952.192\n",
            "  2  GTR+F+I       41420.241    436 83712.482    85436.754    85669.776\n",
            "  3  GTR+F+G4      35411.014    436 71694.028    73418.300    73651.322\n",
            "  4  GTR+F+I+G4    35313.032    437 71500.064    73240.119    73461.847\n",
            "  8  SYM+I+G4      35758.531    434 72385.062    74078.246    74333.377\n",
            " 12  TVM+F+I+G4    35842.463    436 72556.926    74281.198    74514.220\n",
            " 16  TVMe+I+G4     36444.391    433 73754.782    75432.657    75698.608\n",
            " 20  TIM3+F+I+G4   35801.470    435 72472.940    74181.589    74425.744\n",
            " 24  TIM3e+I+G4    37084.913    432 75033.827    76696.547    76973.163\n",
            " 28  TIM2+F+I+G4   35459.803    435 71789.606    73498.255    73742.410\n",
            " 32  TIM2e+I+G4    36570.258    432 74004.516    75667.236    75943.852\n",
            " 36  TIM+F+I+G4    35448.204    435 71766.408    73475.057    73719.212\n",
            " 40  TIMe+I+G4     36279.933    432 73423.866    75086.586    75363.202\n",
            " 44  TPM3u+F+I+G4  36334.740    434 73537.480    75230.664    75485.795\n",
            " 48  TPM3+I+G4     37879.848    431 76621.696    78269.412    78556.543\n",
            " 52  TPM2u+F+I+G4  36010.532    434 72889.064    74582.247    74837.378\n",
            " 56  TPM2+I+G4     37104.656    431 75071.312    76719.028    77006.159\n",
            " 60  K3Pu+F+I+G4   36017.631    434 72903.263    74596.446    74851.578\n",
            " 64  K3P+I+G4      36961.306    431 74784.612    76432.329    76719.460\n",
            " 68  TN+F+I+G4     35885.255    434 72638.511    74331.694    74586.826\n",
            " 72  TNe+I+G4      37659.363    431 76180.727    77828.444    78115.574\n",
            " 76  HKY+F+I+G4    36462.496    433 73790.992    75468.867    75734.818\n",
            " 80  K2P+I+G4      38335.994    430 77531.987    79164.851    79462.345\n",
            " 84  F81+F+I+G4    37473.585    432 75811.170    77473.890    77750.506\n",
            " 88  JC+I+G4       38748.983    429 78355.967    79974.125    80281.836\n",
            "Akaike Information Criterion:           GTR+F+I+G4\n",
            "Corrected Akaike Information Criterion: GTR+F+I+G4\n",
            "Bayesian Information Criterion:         GTR+F+I+G4\n",
            "Best-fit model: GTR+F+I+G4 chosen according to BIC\n",
            "\n",
            "All model information printed to COI_trimmed.fasta.model.gz\n",
            "CPU time for ModelFinder: 46.033 seconds (0h:0m:46s)\n",
            "Wall-clock time for ModelFinder: 45.425 seconds (0h:0m:45s)\n",
            "Generating 1000 samples for ultrafast bootstrap (seed: 724479)...\n",
            "\n",
            "NOTE: 16 MB RAM (0 GB) is required!\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "Thoroughly optimizing +I+G parameters from 10 start values...\n",
            "Init pinv, alpha: 0.000, 0.521 / Estimate: 0.180, 0.477 / LogL: -35327.853\n",
            "Init pinv, alpha: 0.026, 0.521 / Estimate: 0.180, 0.478 / LogL: -35325.783\n",
            "Init pinv, alpha: 0.051, 0.521 / Estimate: 0.181, 0.478 / LogL: -35323.909\n",
            "Init pinv, alpha: 0.077, 0.521 / Estimate: 0.181, 0.478 / LogL: -35321.755\n",
            "Init pinv, alpha: 0.103, 0.521 / Estimate: 0.181, 0.477 / LogL: -35319.589\n",
            "Init pinv, alpha: 0.128, 0.521 / Estimate: 0.182, 0.476 / LogL: -35317.404\n",
            "Init pinv, alpha: 0.154, 0.521 / Estimate: 0.182, 0.475 / LogL: -35315.427\n",
            "Init pinv, alpha: 0.180, 0.521 / Estimate: 0.182, 0.475 / LogL: -35312.994\n",
            "Init pinv, alpha: 0.205, 0.521 / Estimate: 0.183, 0.474 / LogL: -35310.866\n",
            "Init pinv, alpha: 0.231, 0.521 / Estimate: 0.183, 0.473 / LogL: -35308.464\n",
            "Optimal pinv,alpha: 0.183, 0.473 / LogL: -35308.464\n",
            "\n",
            "Parameters optimization took 32.548 sec\n",
            "WARNING: Branch separating outgroup is not found\n",
            "Wrote distance file to... \n",
            "Computing ML distances based on estimated model parameters...\n",
            "Calculating distance matrix: done in 0.24184 secs using 99.53% CPU\n",
            "Computing ML distances took 0.242480 sec (of wall-clock time) 0.241351 sec (of CPU time)\n",
            "WARNING: Some pairwise ML distances are too long (saturated)\n",
            "Setting up auxiliary I and S matrices: done in 0.00178789 secs using 99.61% CPU\n",
            "Computing RapidNJ tree took 0.008131 sec (of wall-clock time) 0.010871 sec (of CPU time)\n",
            "Log-likelihood of RapidNJ tree: -35309.437\n",
            "--------------------------------------------------------------------\n",
            "|             INITIALIZING CANDIDATE TREE SET                      |\n",
            "--------------------------------------------------------------------\n",
            "Generating 98 parsimony trees... 6.618 second\n",
            "Computing log-likelihood of 98 initial trees ... 4.243 seconds\n",
            "Current best score: -35308.464\n",
            "\n",
            "Do NNI search on 20 best initial trees\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 1: -35308.416\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 2: -35005.850\n",
            "Iteration 10 / LogL: -35180.433 / Time: 0h:1m:50s\n",
            "Iteration 20 / LogL: -35180.184 / Time: 0h:2m:9s\n",
            "Finish initializing candidate tree set (20)\n",
            "Current best tree score: -35005.850 / CPU time: 50.059\n",
            "Number of iterations: 20\n",
            "--------------------------------------------------------------------\n",
            "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
            "--------------------------------------------------------------------\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 22: -35000.120\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 25: -34997.312\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 27: -34992.182\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 30: -34989.924\n",
            "Iteration 30 / LogL: -34989.924 / Time: 0h:2m:25s (0h:5m:45s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 33: -34989.715\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 35: -34987.208\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 36: -34968.234\n",
            "Iteration 40 / LogL: -34987.701 / Time: 0h:2m:44s (0h:4m:52s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 41: -34965.449\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 45: -34948.555\n",
            "Iteration 50 / LogL: -34964.323 / Time: 0h:2m:59s (0h:4m:19s left)\n",
            "Log-likelihood cutoff on original alignment: -35156.033\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 58: -34944.430\n",
            "Iteration 60 / LogL: -34965.161 / Time: 0h:3m:15s (0h:4m:9s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 64: -34942.938\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 67: -34934.293\n",
            "Iteration 70 / LogL: -34950.333 / Time: 0h:3m:30s (0h:3m:51s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 75: -34930.409\n",
            "Iteration 80 / LogL: -34941.429 / Time: 0h:3m:45s (0h:3m:37s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 83: -34926.447\n",
            "Iteration 90 / LogL: -34959.567 / Time: 0h:4m:0s (0h:3m:23s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 100: -34925.577\n",
            "Iteration 100 / LogL: -34925.577 / Time: 0h:4m:15s (0h:3m:31s left)\n",
            "Log-likelihood cutoff on original alignment: -35121.744\n",
            "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.878\n",
            "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 103: -34923.728\n",
            "Iteration 110 / LogL: -34938.246 / Time: 0h:4m:29s (0h:3m:11s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 119: -34917.973\n",
            "Iteration 120 / LogL: -34928.829 / Time: 0h:4m:46s (0h:3m:20s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 127: -34917.388\n",
            "Iteration 130 / LogL: -34918.920 / Time: 0h:5m:0s (0h:3m:11s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 138: -34915.844\n",
            "Iteration 140 / LogL: -34917.907 / Time: 0h:5m:14s (0h:3m:9s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 149: -34913.747\n",
            "Iteration 150 / LogL: -34933.296 / Time: 0h:5m:28s (0h:3m:8s left)\n",
            "Log-likelihood cutoff on original alignment: -35121.744\n",
            "Iteration 160 / LogL: -34916.195 / Time: 0h:5m:42s (0h:2m:46s left)\n",
            "Iteration 170 / LogL: -34917.027 / Time: 0h:5m:57s (0h:2m:25s left)\n",
            "Iteration 180 / LogL: -34922.337 / Time: 0h:6m:9s (0h:2m:4s left)\n",
            "Iteration 190 / LogL: -34919.481 / Time: 0h:6m:21s (0h:1m:45s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 192: -34913.348\n",
            "Iteration 200 / LogL: -34927.033 / Time: 0h:6m:35s (0h:2m:41s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.944\n",
            "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 203: -34912.340\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 206: -34909.822\n",
            "Iteration 210 / LogL: -34921.449 / Time: 0h:6m:49s (0h:2m:47s left)\n",
            "Iteration 220 / LogL: -34920.971 / Time: 0h:7m:2s (0h:2m:28s left)\n",
            "Iteration 230 / LogL: -34921.042 / Time: 0h:7m:13s (0h:2m:8s left)\n",
            "Iteration 240 / LogL: -34970.333 / Time: 0h:7m:27s (0h:1m:50s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 247: -34903.396\n",
            "Iteration 250 / LogL: -34920.782 / Time: 0h:7m:40s (0h:2m:41s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "Iteration 260 / LogL: -34906.965 / Time: 0h:7m:53s (0h:2m:23s left)\n",
            "Iteration 270 / LogL: -34914.408 / Time: 0h:8m:7s (0h:2m:6s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 276: -34901.379\n",
            "Iteration 280 / LogL: -34920.442 / Time: 0h:8m:21s (0h:2m:36s left)\n",
            "Iteration 290 / LogL: -34913.074 / Time: 0h:8m:34s (0h:2m:19s left)\n",
            "Iteration 300 / LogL: -34907.343 / Time: 0h:8m:47s (0h:2m:2s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.947\n",
            "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
            "Iteration 310 / LogL: -34945.863 / Time: 0h:9m:4s (0h:2m:25s left)\n",
            "Iteration 320 / LogL: -34919.235 / Time: 0h:9m:17s (0h:2m:8s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 327: -34901.171\n",
            "Iteration 330 / LogL: -34910.666 / Time: 0h:9m:31s (0h:2m:35s left)\n",
            "Iteration 340 / LogL: -34944.340 / Time: 0h:9m:44s (0h:2m:18s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 350: -34900.699\n",
            "Iteration 350 / LogL: -34900.699 / Time: 0h:9m:58s (0h:2m:38s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 353: -34900.682\n",
            "Iteration 360 / LogL: -34906.567 / Time: 0h:10m:12s (0h:2m:26s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 363: -34899.331\n",
            "Iteration 370 / LogL: -34902.400 / Time: 0h:10m:26s (0h:2m:26s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 371: -34898.802\n",
            "Iteration 380 / LogL: -34919.450 / Time: 0h:10m:40s (0h:2m:22s left)\n",
            "Iteration 390 / LogL: -34906.041 / Time: 0h:10m:54s (0h:2m:6s left)\n",
            "Iteration 400 / LogL: -34904.629 / Time: 0h:11m:8s (0h:1m:50s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.937\n",
            "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 405: -34898.018\n",
            "Iteration 410 / LogL: -34904.648 / Time: 0h:11m:22s (0h:2m:28s left)\n",
            "Iteration 420 / LogL: -34907.527 / Time: 0h:11m:35s (0h:2m:11s left)\n",
            "Iteration 430 / LogL: -34902.043 / Time: 0h:11m:49s (0h:1m:56s left)\n",
            "Iteration 440 / LogL: -34902.022 / Time: 0h:12m:3s (0h:1m:40s left)\n",
            "Iteration 450 / LogL: -34908.293 / Time: 0h:12m:15s (0h:1m:24s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "Iteration 460 / LogL: -34901.081 / Time: 0h:12m:31s (0h:1m:9s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 465: -34897.603\n",
            "Iteration 470 / LogL: -34899.172 / Time: 0h:12m:45s (0h:2m:25s left)\n",
            "Iteration 480 / LogL: -34908.153 / Time: 0h:12m:58s (0h:2m:10s left)\n",
            "Iteration 490 / LogL: -34944.486 / Time: 0h:13m:11s (0h:1m:54s left)\n",
            "Iteration 500 / LogL: -34906.685 / Time: 0h:13m:25s (0h:1m:38s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.959\n",
            "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
            "Iteration 510 / LogL: -34911.259 / Time: 0h:13m:40s (0h:2m:17s left)\n",
            "Iteration 520 / LogL: -34910.441 / Time: 0h:13m:54s (0h:2m:1s left)\n",
            "Iteration 530 / LogL: -34901.714 / Time: 0h:14m:8s (0h:1m:46s left)\n",
            "Iteration 540 / LogL: -34910.819 / Time: 0h:14m:22s (0h:1m:30s left)\n",
            "Iteration 550 / LogL: -34905.039 / Time: 0h:14m:38s (0h:1m:15s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "Iteration 560 / LogL: -34905.493 / Time: 0h:14m:53s (0h:1m:0s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 569: -34897.343\n",
            "Iteration 570 / LogL: -34905.523 / Time: 0h:15m:6s (0h:2m:29s left)\n",
            "Iteration 580 / LogL: -34904.584 / Time: 0h:15m:20s (0h:2m:14s left)\n",
            "Iteration 590 / LogL: -34903.834 / Time: 0h:15m:34s (0h:1m:59s left)\n",
            "Iteration 600 / LogL: -34943.647 / Time: 0h:15m:48s (0h:1m:44s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.989\n",
            "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
            "Iteration 610 / LogL: -34899.381 / Time: 0h:16m:3s (0h:2m:15s left)\n",
            "Iteration 620 / LogL: -34900.377 / Time: 0h:16m:17s (0h:2m:0s left)\n",
            "Iteration 630 / LogL: -34902.559 / Time: 0h:16m:32s (0h:1m:45s left)\n",
            "Iteration 640 / LogL: -34903.097 / Time: 0h:16m:46s (0h:1m:30s left)\n",
            "Estimate model parameters (epsilon = 0.100)\n",
            "BETTER TREE FOUND at iteration 647: -34896.251\n",
            "Iteration 650 / LogL: -34908.676 / Time: 0h:17m:0s (0h:2m:25s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "Iteration 660 / LogL: -34904.567 / Time: 0h:17m:16s (0h:2m:10s left)\n",
            "Iteration 670 / LogL: -34901.453 / Time: 0h:17m:30s (0h:1m:55s left)\n",
            "Iteration 680 / LogL: -34901.176 / Time: 0h:17m:46s (0h:1m:40s left)\n",
            "Iteration 690 / LogL: -34917.931 / Time: 0h:18m:1s (0h:1m:25s left)\n",
            "Iteration 700 / LogL: -34918.892 / Time: 0h:18m:17s (0h:1m:10s left)\n",
            "Log-likelihood cutoff on original alignment: -35003.992\n",
            "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.993\n",
            "Iteration 710 / LogL: -34908.377 / Time: 0h:18m:32s (0h:2m:15s left)\n",
            "Iteration 720 / LogL: -34913.863 / Time: 0h:18m:47s (0h:2m:0s left)\n",
            "Iteration 730 / LogL: -34919.609 / Time: 0h:19m:2s (0h:1m:45s left)\n",
            "Iteration 740 / LogL: -34903.784 / Time: 0h:19m:17s (0h:1m:30s left)\n",
            "TREE SEARCH COMPLETED AFTER 748 ITERATIONS / Time: 0h:19m:29s\n",
            "\n",
            "--------------------------------------------------------------------\n",
            "|                    FINALIZING TREE SEARCH                        |\n",
            "--------------------------------------------------------------------\n",
            "Performs final model parameters optimization\n",
            "Estimate model parameters (epsilon = 0.010)\n",
            "1. Initial log-likelihood: -34896.251\n",
            "Optimal log-likelihood: -34896.250\n",
            "Rate parameters:  A-C: 3.31727  A-G: 10.91912  A-T: 15.58074  C-G: 2.13904  C-T: 63.19989  G-T: 1.00000\n",
            "Base frequencies:  A: 0.304  C: 0.158  G: 0.146  T: 0.392\n",
            "Proportion of invariable sites: 0.182\n",
            "Gamma shape alpha: 0.467\n",
            "Parameters optimization took 1 rounds (0.152 sec)\n",
            "BEST SCORE FOUND : -34896.250\n",
            "Creating bootstrap support values...\n",
            "Split supports printed to NEXUS file COI_trimmed.fasta.splits.nex\n",
            "Total tree length: 61.946\n",
            "\n",
            "Total number of iterations: 748\n",
            "CPU time used for tree search: 1071.707 sec (0h:17m:51s)\n",
            "Wall-clock time used for tree search: 1090.348 sec (0h:18m:10s)\n",
            "Total CPU time used: 1152.308 sec (0h:19m:12s)\n",
            "Total wall-clock time used: 1170.623 sec (0h:19m:30s)\n",
            "\n",
            "Computing bootstrap consensus tree...\n",
            "Reading input file COI_trimmed.fasta.splits.nex...\n",
            "215 taxa and 2302 splits.\n",
            "Consensus tree written to COI_trimmed.fasta.contree\n",
            "Reading input trees file COI_trimmed.fasta.contree\n",
            "Log-likelihood of consensus tree: -34899.875\n",
            "WARNING: Branch separating outgroup is not found\n",
            "WARNING: Branch separating outgroup is not found\n",
            "\n",
            "Analysis results written to: \n",
            "  IQ-TREE report:                COI_trimmed.fasta.iqtree\n",
            "  Maximum-likelihood tree:       COI_trimmed.fasta.treefile\n",
            "  Likelihood distances:          COI_trimmed.fasta.mldist\n",
            "\n",
            "Ultrafast bootstrap approximation results written to:\n",
            "  Split support values:          COI_trimmed.fasta.splits.nex\n",
            "  Consensus tree:                COI_trimmed.fasta.contree\n",
            "  Screen log file:               COI_trimmed.fasta.log\n",
            "\n",
            "Date and Time: Sun Oct 27 14:09:25 2024\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## **Step 3.5 Rename the species**\n",
        "\n",
        "Go to \"Downloading_data\" colab."
      ],
      "metadata": {
        "id": "17ohMAl22hMw"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## **Step 4. Tree visualisation**\n",
        "\n",
        "We are going to use [ITOl](https://itol.embl.de/) to do it."
      ],
      "metadata": {
        "id": "_85QYH472Bix"
      }
    }
  ]
}